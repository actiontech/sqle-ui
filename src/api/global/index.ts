/* tslint:disable no-identical-functions */
/* tslint:disable no-useless-cast */
/* tslint:disable no-unnecessary-type-assertion */
/* tslint:disable no-big-function  */
/* tslint:disable no-duplicate-string  */
import ServiceBase from '../Service.base';
import { AxiosRequestConfig } from 'axios';

import { IGetSQLEInfoReturn } from './index.d';

class GlobalService extends ServiceBase {
  public getSQLEInfo(options?: AxiosRequestConfig) {
    return this.get<IGetSQLEInfoReturn>('/v1/basic_info', undefined, options);
  }
}

export default new GlobalService();
