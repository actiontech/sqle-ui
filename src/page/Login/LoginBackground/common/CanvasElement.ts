import { IPosition } from './index.type';

abstract class CanvasElement {
  abstract position: IPosition;
  abstract canvas: HTMLCanvasElement;
  abstract context: CanvasRenderingContext2D;
  abstract draw(): void;
  abstract update(reachPosition: IPosition): void;
}

export default CanvasElement;
