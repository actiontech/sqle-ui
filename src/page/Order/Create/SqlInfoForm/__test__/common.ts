import instance from '../../../../../api/instance';
import { resolveThreeSecond } from '../../../../../testUtils/mockRequest';

export const mockGetInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceV2');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      instance_name: 'db1',
      db_host: '20.20.20.2',
      db_port: '3306',
      db_user: 'root',
      db_type: 'mysql',
      desc: '',
      workflow_template_name: 'workflow-template-name-1',
      rule_template: {
        name: 'not_submit_test_rule33',
        is_global_rule_template: true,
      },
    })
  );

  return spy;
};
