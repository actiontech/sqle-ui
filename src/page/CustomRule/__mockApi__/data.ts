import { ICustomRuleResV1 } from '../../../api/common';
import { CustomRuleResV1LevelEnum } from '../../../api/common.enum';

export const customRules: ICustomRuleResV1[] = [
  {
    db_type: 'mysql',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.notice,
    rule_id: '1',
    desc: 'name1',
    rule_script: 'w+',
    type: 'rule_type1',
  },
  {
    db_type: 'oracle',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.error,
    rule_id: '2',
    desc: 'name2',
    rule_script: 'w+',
    type: 'rule_type2',
  },
];
