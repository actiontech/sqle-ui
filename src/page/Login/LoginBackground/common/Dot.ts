import { random } from '../../../../utils/Math';
import CanvasElement from './CanvasElement';
import { IPosition } from './index.type';

class Dot extends CanvasElement {
  public canvas: HTMLCanvasElement;
  public context: CanvasRenderingContext2D;
  public position: IPosition = { x: 0, y: 0 };
  private size: number;
  private color: string;
  private slope!: number;
  private offset!: number;
  private speed: number;
  private randomNum: number;
  private autoMove: boolean = true;

  constructor(
    canvas: HTMLCanvasElement,
    speed?: number,
    size?: number,
    color?: string,
    slope?: number,
    offset?: number
  ) {
    super();
    this.canvas = canvas;
    this.context = canvas.getContext('2d') as CanvasRenderingContext2D;
    this.size = size ?? 3;
    this.color = color ?? '#fff';
    this.speed = speed ?? 1;
    this.randomNum = Math.random() * 5 + 5;
    this.initPosition();
    if (!slope || !offset) {
      this.randomTrajectory();
    }
  }

  public draw(): void {
    this.context.beginPath();
    this.context.fillStyle = this.color;
    this.context.arc(
      this.position.x,
      this.position.y,
      this.size,
      0,
      Math.PI * 2
    );
    this.context.fill();
    this.context.closePath();
  }

  public update(reachPosition?: IPosition): void {
    let { x, y } = this.position;
    if (!reachPosition) {
      if (!this.autoMove) {
        return;
      }
      x += this.speed;
      if (x <= 0 || x >= this.canvas.width) {
        x = 0;
      }
      y = this.slope * x + this.offset;
    } else {
      x = x + ((reachPosition.x - x) * this.randomNum) / 50;
      y = y + ((reachPosition.y - y) * this.randomNum) / 50;
    }
    this.position = { x, y };
  }

  public closeAutoMove() {
    this.autoMove = false;
  }

  public openAutoMove() {
    this.randomTrajectory();
    this.autoMove = true;
  }

  public randomTrajectory() {
    const slopeBig = random(1, this.canvas.height / 2000);
    const slopeSmall = Math.random();
    const direction = Math.random() > 0.5 ? 1 : -1;
    const slope = (Math.random() > 0.5 ? slopeBig : slopeSmall) * direction;
    const offset = this.position.y - slope * this.position.x;
    this.slope = slope;
    this.offset = offset;
  }

  public initPosition() {
    this.position = {
      x: random(1, this.canvas.width - 1),
      y: random(1, this.canvas.height - 1),
    };
  }
}

export default Dot;
