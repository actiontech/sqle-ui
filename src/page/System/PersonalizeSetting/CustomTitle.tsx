import { EditOutlined, EnterOutlined } from '@ant-design/icons';
import { useBoolean, useKeyPress } from 'ahooks';
import { Col, Input, message, Row, Typography } from 'antd';
import { useEffect, useRef, useState } from 'react';
import { useTranslation } from 'react-i18next';
import configuration from '../../../api/configuration';
import EmptyBox from '../../../components/EmptyBox';
import { ResponseCode } from '../../../data/common';

const CustomTitle: React.FC<{
  title: string;
  refresh: () => void;
}> = ({ title, refresh }) => {
  const { t } = useTranslation();
  const [editTitle, { setTrue: startEditTitle, setFalse: closeEditTitle }] =
    useBoolean(false);
  const inputRef = useRef<Input | null>(null);
  const [inputValue, setInputValue] = useState('');
  const updateTitle = async (event: KeyboardEvent) => {
    const value = (event.target as HTMLInputElement).value;
    if (value === title) {
      message.error(t('system.personalize.match'));
      return;
    }

    const res = await configuration.personalise({ title: value });
    if (res.data.code === ResponseCode.SUCCESS) {
      message.success(t('system.personalize.updateTitleSuccessTips'));
      refresh();
    }

    closeEditTitle();
  };

  const inputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(event.target.value);
  };

  useKeyPress('esc', closeEditTitle, {
    target: () => document.getElementById('titleInput'),
  });

  useKeyPress('enter', updateTitle, {
    target: () => document.getElementById('titleInput'),
  });

  useEffect(() => {
    if (editTitle) {
      inputRef.current?.focus();
    }
  }, [editTitle]);
  return (
    <Row align="middle">
      <Col sm={3} xl={2} xs={24}>
        <Typography.Text>{t('system.personalize.title')} : </Typography.Text>
      </Col>
      <Col sm={10} xs={24}>
        <div hidden={editTitle}>
          <Typography.Text>
            <EmptyBox if={!!title} defaultNode="--">
              {title}
            </EmptyBox>
            <Typography.Link onClick={startEditTitle}>
              <EditOutlined />
            </Typography.Link>
          </Typography.Text>
        </div>
        <div hidden={!editTitle}>
          <Input
            id="titleInput"
            ref={(ref) => (inputRef.current = ref)}
            suffix={<EnterOutlined />}
            value={inputValue}
            onChange={inputChange}
            onBlur={closeEditTitle}
            placeholder={t('common.form.placeholder.input', {
              name: t('system.personalize.title'),
            })}
          />
        </div>
      </Col>
    </Row>
  );
};

export default CustomTitle;
