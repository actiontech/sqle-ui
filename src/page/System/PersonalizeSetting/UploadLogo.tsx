import { LoadingOutlined, UploadOutlined } from '@ant-design/icons';
import { useBoolean } from 'ahooks';
import { Col, Row, Space, Typography, Upload, UploadProps } from 'antd';
import { useTranslation } from 'react-i18next';
import configuration from '../../../api/configuration';
import { ResponseCode } from '../../../data/common';
import logo from '../../../assets/img/logo.png';

const UploadLogo: React.FC<{ url?: string; refresh: () => void }> = ({
  url,
  refresh,
}) => {
  const { t } = useTranslation();
  const [loading, { setFalse: finishUpload, setTrue: startUpload }] =
    useBoolean(false);

  const customRequest: UploadProps['customRequest'] = (options) => {
    startUpload();
    configuration
      .uploadLogo(
        {
          logo: options.file,
        },
        {
          headers: {
            'content-type': 'multipart/form-data',
          },
        }
      )
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          window.location.reload();
          refresh();
        }
      })
      .finally(() => {
        finishUpload();
      });
  };

  return (
    <Row>
      <Col sm={3} xl={2} xs={24}>
        <Typography.Text>{t('system.personalize.logo')} : </Typography.Text>
      </Col>
      <Col sm={10} xs={24}>
        <Space align="end">
          <Upload
            data-testid="upload-logo"
            name="logo"
            listType="picture-card"
            showUploadList={false}
            customRequest={customRequest}
            disabled={loading}
          >
            {loading ? (
              <LoadingOutlined />
            ) : (
              <Space direction="vertical">
                <img
                  style={{
                    width: '50%',
                  }}
                  src={url ? url : logo}
                  alt="logo"
                />
                <Typography.Text type="secondary">
                  <UploadOutlined />
                  {t('system.personalize.uploadAndUpdate')}
                </Typography.Text>
              </Space>
            )}
          </Upload>
          <Typography.Text type="secondary">
            {t('system.personalize.uploadTips')}
          </Typography.Text>
        </Space>
      </Col>
    </Row>
  );
};

export default UploadLogo;
