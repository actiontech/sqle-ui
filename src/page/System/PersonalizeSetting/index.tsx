import { useRequest } from 'ahooks';
import { Card, Space } from 'antd';
import { useTranslation } from 'react-i18next';
import CustomTitle from './CustomTitle';
import GlobalService from '../../../api/global';
import { SQLE_DEFAULT_WEB_TITLE } from '../../../data/common';
import UploadLogo from './UploadLogo';
import { useDispatch } from 'react-redux';
import { updateWebTitleAndLog } from '../../../store/system';

const PersonalizeSetting: React.FC = () => {
  const { t } = useTranslation();
  const dispatch = useDispatch();

  const {
    data: sqleInfo,
    loading,
    refresh,
  } = useRequest(() =>
    GlobalService.getSQLEInfoV1().then((res) => {
      const webTitle = res.data.data?.title ?? SQLE_DEFAULT_WEB_TITLE;
      document.title = webTitle;
      dispatch(
        updateWebTitleAndLog({
          webTitle,
          webLogoUrl: res.data.data?.logo_url,
        })
      );
      return res.data.data;
    })
  );

  return (
    <Card loading={loading} title={t('system.title.personalize')}>
      <Space direction="vertical" className="full-width-element">
        <CustomTitle
          title={sqleInfo?.title ?? SQLE_DEFAULT_WEB_TITLE}
          refresh={refresh}
        />

        <UploadLogo url={sqleInfo?.logo_url} refresh={refresh} />
      </Space>
    </Card>
  );
};

export default PersonalizeSetting;
