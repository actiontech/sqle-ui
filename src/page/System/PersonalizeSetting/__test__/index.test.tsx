import { render, act } from '@testing-library/react';
import PersonalizeSetting from '..';
import global from '../../../../api/global';
import { SQLE_DEFAULT_WEB_TITLE } from '../../../../data/common';
import { resolveThreeSecond } from '../../../../testUtils/mockRequest';
import { useDispatch } from 'react-redux';

const mockGetSqleInfo = () => {
  const spy = jest.spyOn(global, 'getSQLEInfoV1');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      title: 'SQLE',
      logo_url: 'test',
    })
  );
  return spy;
};

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useDispatch: jest.fn(),
  };
});

describe('test PersonalizeSetting', () => {
  let getSqleInfoSpy: jest.SpyInstance;
  const dispatchSpy = jest.fn();

  beforeEach(() => {
    jest.useFakeTimers();
    getSqleInfoSpy = mockGetSqleInfo();
    (useDispatch as jest.Mock).mockImplementation(() => dispatchSpy);
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    jest.clearAllTimers();
  });
  test('should match snapshot', async () => {
    const { container } = render(<PersonalizeSetting />);
    await act(async () => jest.advanceTimersByTime(3000));

    expect(container).toMatchSnapshot();
  });

  test('should dispatch "updateWebTitleAndLog" action and set document title after getting sqle data from the request', async () => {
    render(<PersonalizeSetting />);
    expect(getSqleInfoSpy).toBeCalledTimes(1);
    await act(async () => jest.advanceTimersByTime(3000));

    expect(document.title).toBe('SQLE');
    expect(dispatchSpy).toBeCalledTimes(1);
    expect(dispatchSpy).toBeCalledWith({
      payload: {
        webLogoUrl: 'test',
        webTitle: 'SQLE',
      },
      type: 'system/updateWebTitleAndLog',
    });
  });

  test('should set default title when the fetched title is undefined', async () => {
    getSqleInfoSpy.mockImplementation(() => resolveThreeSecond({}));
    render(<PersonalizeSetting />);
    await act(async () => jest.advanceTimersByTime(3000));

    expect(document.title).toBe(SQLE_DEFAULT_WEB_TITLE);
  });
});
