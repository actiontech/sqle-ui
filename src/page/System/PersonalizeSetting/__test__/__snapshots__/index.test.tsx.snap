// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PersonalizeSetting should match snapshot 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          system.title.personalize
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-space ant-space-vertical full-width-element"
      >
        <div
          class="ant-space-item"
          style="margin-bottom: 8px;"
        >
          <div
            class="ant-row ant-row-middle"
          >
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-3 ant-col-xl-2"
            >
              <span
                class="ant-typography"
              >
                system.personalize.title
                 : 
              </span>
            </div>
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-10"
            >
              <div>
                <span
                  class="ant-typography"
                >
                  SQLE
                  <a
                    class="ant-typography"
                  >
                    <span
                      aria-label="edit"
                      class="anticon anticon-edit"
                      role="img"
                      tabindex="-1"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="edit"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                        />
                      </svg>
                    </span>
                  </a>
                </span>
              </div>
              <div
                hidden=""
              >
                <span
                  class="ant-input-affix-wrapper"
                >
                  <input
                    class="ant-input"
                    id="titleInput"
                    placeholder="common.form.placeholder.input"
                    type="text"
                    value=""
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span
                      aria-label="enter"
                      class="anticon anticon-enter"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="enter"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M864 170h-60c-4.4 0-8 3.6-8 8v518H310v-73c0-6.7-7.8-10.5-13-6.3l-141.9 112a8 8 0 000 12.6l141.9 112c5.3 4.2 13 .4 13-6.3v-75h498c35.3 0 64-28.7 64-64V178c0-4.4-3.6-8-8-8z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-space-item"
        >
          <div
            class="ant-row"
          >
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-3 ant-col-xl-2"
            >
              <span
                class="ant-typography"
              >
                system.personalize.logo
                 : 
              </span>
            </div>
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-10"
            >
              <div
                class="ant-space ant-space-horizontal ant-space-align-end"
              >
                <div
                  class="ant-space-item"
                  style="margin-right: 8px;"
                >
                  <span
                    class="ant-upload-picture-card-wrapper"
                  >
                    <div
                      class="ant-upload ant-upload-select ant-upload-select-picture-card"
                    >
                      <span
                        class="ant-upload"
                        role="button"
                        tabindex="0"
                      >
                        <input
                          accept=""
                          data-testid="upload-logo"
                          style="display: none;"
                          type="file"
                        />
                        <div
                          class="ant-space ant-space-vertical"
                        >
                          <div
                            class="ant-space-item"
                            style="margin-bottom: 8px;"
                          >
                            <img
                              alt="logo"
                              src="test"
                              style="width: 50%;"
                            />
                          </div>
                          <div
                            class="ant-space-item"
                          >
                            <span
                              class="ant-typography ant-typography-secondary"
                            >
                              <span
                                aria-label="upload"
                                class="anticon anticon-upload"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="upload"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                                  />
                                </svg>
                              </span>
                              system.personalize.uploadAndUpdate
                            </span>
                          </div>
                        </div>
                      </span>
                    </div>
                  </span>
                </div>
                <div
                  class="ant-space-item"
                >
                  <span
                    class="ant-typography ant-typography-secondary"
                  >
                    system.personalize.uploadTips
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
